server/graphql/v1/mutations/notifications.js

```diff
@@ -0,0 +1,88 @@
+import Promise from 'bluebird';
+import { values } from 'lodash';
+import models, { Op } from '../../../models';
+import activities from '../../../constants/activities';
+import { channels } from '../../../constants';
+import { Forbidden } from '../../errors';
+
+const NotificationPermissionError = new Forbidden({
+  message: "This notification does not exist or you don't have the permission to edit it.",
+});
+
+export async function editNotifications(args, remoteUser) {
+  if (!remoteUser) {
+    throw NotificationPermissionError;
+  }
+
+  if (!args.notifications) return Promise.resolve();
+  let newNotifications = [];
+
+  return models.Notification.findAll({ where: { CollectiveId: args.id, UserId: null } })
+    .then(oldNotifications => {
+      const diff = oldNotifications
+        .filter(
+          x1 =>
+            !args.notifications.some(x2 => {
+              return x2.channel === x1.channel && x2.type === x1.type && x2.webhookUrl === x1.webhookUrl;
+            }),
+        )
+        .map(x => x.id);
+
+      newNotifications = args.notifications.filter(
+        x1 =>
+          !oldNotifications.some(x2 => {
+            return x2.channel === x1.channel && x2.type === x1.type && x2.webhookUrl === x1.webhookUrl;
+          }),
+      );
+
+      return models.Notification.destroy({ where: { id: { [Op.in]: diff } } });
+    })
+    .then(() => {
+      return Promise.map(newNotifications, notification => {
+        if (!(values(activities).includes(notification.type) && values(channels).includes(notification.channel)))
+          return;
+
+        notification.CollectiveId = args.id;
+        return models.Notification.create(notification);
+      });
+    });
+}
+
+/**
+ * Creates a new notification
+ */
+export async function createNotification(args, remoteUser) {
+  if (!remoteUser) {
```

Comment:
We also need to check that `remoteUser` is admin of collective `CollectiveId`. See https://github.com/opencollective/opencollective-api/blob/32f473bdbd17056bf8dbef7e6e6c392c32fa8747/server/graphql/v1/mutations/collectives.js#L691 for an example

Reply:
I think we want non-admins to be able to register event webhooks as well?

Reply:
I'll let @znarf confirm that as I don't know how the frontend implementation is going to look like. If webhooks are meant to be created from the admin interface ("edit collective") then we should be protective per default.

Reply:
As this lives in the `Edit collective` menu that is only accessible to Collective admins, I can confirm that we should check for `req.remoteUser.isAdmin(CollectiveId)`

Reply:
This `createNotification` mutation is supposed to be called by Zapier, and we want non-admins to integrate via Zapier as well.

The `Edit collective` menu calls the `editNotifications` mutation which is defined above.

A difference between the two mutations is that `createNotification` persists the user ID as well, while `editNotifications` does not because the latter is supposed to be collective-wide, while the former can be integrated by anybody.

Reply:
My bad, it's all good then. Thanks for sharing the details.

Reply:
@eloyekunle As discussed on Slack, this is where we should ensure that user is admin of the collective, otherwise anyone could create a webhook to a malicious endpoint and make private data leak.

Reply:
@Betree I have updated the mutations to check admin status.